dev/benchmarks/macrobenchmarks/lib/src/color_filter_and_fade.dart

```diff
@@ -0,0 +1,176 @@
+          row,
+          const SizedBox(height: 12),
+          row,
+          const SizedBox(height: 12),
+        ],
+    );
+
+    final Widget fadeTransition = FadeTransition(
+      opacity: _opacityAnimation,
+      // This RepaintBoundary is necessary to not let the opacity change
+      // invalidate the layer raster cache below. This is necessary with
+      // or without the color filter.
+      child: RepaintBoundary(
+        child: column,
+      ),
+    );
+
+    return Scaffold(
+        backgroundColor: Colors.lightBlue,
+        body: Center(
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                fadeTransition,
+                Container(height: 20),
+                const Text('Use Color Filter:'),
+                Checkbox(
+                  value: _useColorFilter,
+                  onChanged: (bool value) {
+                    setState(() {
+                      _useColorFilter = value;
+                    });
+                  },
+                ),
+              ],
+            ),
+        ),
+    );
+  }
+
+  // Create a looping fade-in fade-out animation for opacity.
+  void _initAnimation() {
+    _controller = AnimationController(duration: const Duration(seconds: 3), vsync: this);
+    _opacityAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(_controller);
+    _opacityAnimation.addStatusListener((AnimationStatus status) {
+      if (status == AnimationStatus.completed) {
+        _controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        _controller.forward();
+      }
+    });
+    _controller.forward();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _initAnimation();
+  }
+
+  AnimationController _controller;
+  Animation<double> _opacityAnimation;
+  bool _useColorFilter = true;
+}
+
+class _ShadowWidget extends StatelessWidget {
+  const _ShadowWidget({
+    @required this.width,
+    @required this.height,
+    @required this.color,
+    @required this.shadow,
+  });
+
+  final double width;
+  final double height;
+  final Color color;
+  final Shadow shadow;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: width,
+      height: height,
+      child: CustomPaint(
+        painter: _ShadowPainter(
+          color: color,
+          shadow: shadow,
+        ),
+        size: Size(width, height),
+        isComplex: true,
+        willChange: false,
+      ),
+    );
+  }
+}
+
+class _ShadowPainter extends CustomPainter {
+  const _ShadowPainter({this.color, @required this.shadow});
+
+  // If null, we'll set no color filter.
+  final Color color;
+
+  final Shadow shadow;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Rect rect = Offset.zero & size;
+
+    final Paint paint = Paint();
+    if (color != null) {
+      paint.colorFilter = ColorFilter.mode(shadow.color, BlendMode.srcIn);
```

Comment:
You are testing color for null, but then using shadow.color here. Does color get used as anything other than a flag?